processing.go

```diff
@@ -24,6 +26,11 @@ type ScanTarget struct {
 	Domain string
 }
 
+// ScanTarget.Open connects to the ScanTarget using net.DialTimeout(), getting the port and timeout from flags.
+func (t *ScanTarget) Open(flags *BaseFlags) (net.Conn, error) {
+	return net.DialTimeout("tcp", fmt.Sprintf("%s:%d", t.IP.String(), flags.Port), time.Second*time.Duration(flags.Timeout))
```

Comment:
I'm wondering if it would be feasible to use this to automatically set the IO timeouts as well -- i.e. return a new net.Conn implementation (wrapping the `net.DialTimeout()` return value) that calls `SetReadDeadline()` etc, since currently that needs to be re-implemented everywhere.

The main risk I see is, some scanners may need more flexibility (wait different times for different ops) -- but in that case, there's nothing preventing anyone from calling net.DialTimeout directly.